api/Dockerfile

```diff
@@ -19,22 +25,10 @@ RUN adduser \
     --uid "${UID}" \
     "${USER}"
 
-
-WORKDIR /app
-
-# First we handle fetching and building our dependencies
-# We need a stub entry point for that
-COPY Cargo.toml Cargo.lock ./
-COPY ./src/stub.rs ./src/main.rs
-RUN cargo fetch --target x86_64-unknown-linux-musl
-RUN cargo build --offline --target x86_64-unknown-linux-musl --release
-
-COPY ./ .
+# Build release
```

Comment:
Can you keep the 2 steps build system? It is an important optimization that is allowing us to iterate faster.
Building all dependencies currently takes ~2min and it could go up when we add more. With your change we will have to pay for a full compilation each time we start the composition.
By building the dependencies separately it only takes 5s to build the api.

Reply:
Fetching & building is done in the dev container when the `watch` command is invoked through docker-compose. It should reuse the image with all the cache and facilitate incremental builds between docker-compose runs.

Reply:
Will test when the local dev works again.
Since you removed the `--offline` parameter rust will regularily spend time re-downloading the crates index, right?

Reply:
I added the offline flag here as it only applies to a docker build anyways. What do you mean by `local dev`? You can test this by removing & purging everything from previous composition runs and the just use `make run`.

Reply:
I mean dev on the developer machine. I just pulled and tried a `make run` but it doesn't rebuild the image so I can not test. (see my comment above about the issue).

Reply:
I did nothing special since I wanted to test that "make run" automatically take changes in the project and brings up an up-to-date composition.

Reply:
This does not work with the changes made here due to docker-compose limitations.